### ISSUE TITLE
Wire.requestFrom() requests 4 bytes instead of 1.

### ISSUE THREAD

**Issue by Koepel** [Coolbots7/BatteryManager#2]
In the files:
- BatteryManager/ArduinoLibrary/BatteryManager/BatteryManager.cpp
- BatteryManager/I2C_Test/I2C_Test.ino

In the function "getByte()", the Wire.requestFrom() requests 4 bytes (the size of a float), but it could request just one byte. I think you need to cast it with `(uint8_t) 1`.

**Comment by Coolbots7** [comment]
Good catch, thanks.

I think it was introduced when I copied the getFloat() function which getByte() was probably based on.

I plan on overhauling some parts of the code and it will be handled then.

**Comment by Koepel** [comment]
I noticed a timeout in "BatteryManager/src/ArduinoLibrary/BatteryManager/BatteryManager.cpp" after a Wire.requestFrom(). Was that timeout already there and did I miss that ?
There is no such thing as a timeout after Wire.requestFrom().
Explanation: [Common-mistakes#1](https://github.com/Koepel/How-to-use-the-Arduino-Wire-library/wiki/Common-mistakes#1)

You can replace this:
```
  unsigned long wait_start = millis();
  while ( (Wire.available() <= 0) && (millis() - wait_start < REQUEST_TIMEOUT));
  if(millis() - wait_start >= REQUEST_TIMEOUT) {
    Serial.println("I2C REQUEST TIMEOUT");
    return;
  }

  num_bytes = Wire.available();
  byte received_bytes[num_bytes];
```
with this:
```
  if (num_bytes != Wire.available())
  {
    Serial.println("I2C REQUEST ERROR");
    return;
  }
  byte received_bytes[num_bytes];
```

**Comment by Coolbots7** [comment]
Thanks, resolved in c277b2f
